integration-tests/vertx-graphql/src/test/java/io/quarkus/vertx/graphql/it/VertxGraphqlTest.java

```diff
@@ -20,10 +18,6 @@
 
 @QuarkusTest
 class VertxGraphqlTest {
-
-    @Inject
-    Vertx vertx;
```

Comment:
This injection works in JVM mode but not in native?

Reply:
Yes, the injection is not working in native. I am entirely not sure why though as could not get to the bottom of it. 

@mkouba @manovotn any ideas?

Reply:
Do you have anything more specific? What was the exception or the stack trace?
Also where do we actually produce the `Vertx` instance? I can see [`VertxProducer ` but that one is already injecting it,](https://github.com/quarkusio/quarkus/blob/master/extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxProducer.java#L20-L21) so it has to be elsewhere.

Reply:
@cristianonicolai we are trying to figure out why we have a `null` injection of Vertx in native mode. Any insights is welcomed.

Reply:
> Do you have anything more specific? What was the exception or the stack trace?

It works correctly in JVM test but the value is `null` in native test. No any specific CDI stacktrace other than a NPE thrown here https://github.com/quarkusio/quarkus/pull/5253/files#diff-689b6ba08caa09cd266fc629886bcba3L40

Reply:
Oh, so just null. Hmm,then we probably need to figure out how is that bean registered in the first place.

Reply:
The bean is produced here - https://github.com/quarkusio/quarkus/blob/f98471dfb7dc0d26918a7d817b7360530d65cdd1/extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreProducer.java

I am able to reproduce in another native test e.g adding the following code for
```
   @Inject
    Flyway flyway;

    @Test
    public void test() {
        System.out.println(flyway);
        System.out.println(Arc.container());
    }
```
in https://github.com/quarkusio/quarkus/blob/785b0d5b904fe74e210084aec0d4c8448d27f7f0/integration-tests/flyway/src/test/java/io/quarkus/it/flyway/FlywayFunctionalityNativeIT.java 

Will print 
```
null
null
```

Note that `ArcContainer` is `null`, meaning not initialised at all in native test? `

Reply:
@machi1990 after reading the docs I got the impression that injection isnt supported in native tests at all https://quarkus.io/guides/getting-started-testing#native-executable-testing

Reply:
Good catch @cristianonicolai thanks.  I was in a point of opening a separate issue for that.

Reply:
Yes, injection is not supported in `@NativeImageTest` (and it's mentioned in the [guide](https://quarkus.io/guides/getting-started-testing#native-executable-testing)). The reason is that the native executable runs in a separate process and the test class is of course not part of the native image.